### PR TITLE
graphlog: add `enum GraphStyle` to replace string parameter

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: cli/tests/test_log_command.rs [hunk 1]
```diff
@@ -1280,6 +1280,16 @@ fn test_graph_styles() {
...
     "###);
+
+    // Invalid style name
+    let stderr = test_env.jj_cmd_failure(
+        &repo_path,
+        &["log", "--config-toml=ui.graph.style='unknown'"],
+    );
+    insta::assert_snapshot!(stderr, @r###"
+    Config error: enum GraphStyle does not have variant constructor unknown
+    For help, see https://github.com/martinvonz/jj/blob/main/docs/config.md.
```
Comment (martinvonz) [inline cli/tests/test_log_command.rs:1291]: Should this point to https://martinvonz.github.io/jj/latest/config/ instead?

File: cli/src/graphlog.rs [hunk 2]
```diff
@@ -132,12 +130,12 @@ pub fn node_template_for_key(
     key: &str,
     fallback: &str,
     ascii_fallback: &str,
-) -> String {
-    let symbol = settings.config().get_string(key);
-    if GraphStyle::from_settings(settings).is_ascii() {
-        symbol.unwrap_or_else(|_| ascii_fallback.to_owned())
+) -> Result<String, config::ConfigError> {
```
Comment (martinvonz) [inline cli/src/graphlog.rs]: Do you think it's worth mentioning this in the changelog as a breaking change?

**Review by martinvonz — APPROVED** [review]

File: cli/tests/test_log_command.rs (hunk 1 above)
Comment (yuja) [reply cli/tests/test_log_command.rs:1291]: Done in https://github.com/martinvonz/jj/pull/4393

File: cli/src/graphlog.rs (hunk 2 above)
Comment (yuja) [reply cli/src/graphlog.rs]: Added changelog entry.

**Pull request merged**